rtos/ConditionVariable.h

```diff
@@ -65,78 +83,98 @@ struct Waiter;
  * @note Synchronization level: Thread safe
  *
  * Example:
+ *
  * @code
  * #include "mbed.h"
  *
  * Mutex mutex;
- * ConditionVariable cond(mutex);
+ * ConditionVariable cv(mutex);
  *
  * // These variables are protected by locking mutex
- * uint32_t count = 0;
+ * uint32_t work_count = 0;
  * bool done = false;
  *
  * void worker_thread()
  * {
- *     mutex.lock();
- *     do {
- *         printf("Worker: Count %lu\r\n", count);
+ *   // Acquire lock on mutex before accessing protected variables and waiting.
+ *   mutex.lock();
  *
- *         // Wait for a condition to change
- *         cond.wait();
+ *   while (done == false) {
```

Comment:
Good point on making this a `while` loop. `do` is pretty dangerous with a condition variable.

But ` == false` or `== true` is bad style

Either `done` isn't a boolean, in which case it's actively dangerous, or it is a boolean, in which case it's just obfuscation.

`if (done)` or `while (!done)` on a boolean reads more like natural language.